framework/core/src/Discussion/Discussion.php

```diff
@@ -117,7 +117,7 @@ public static function boot()
     /**
      * Start a new discussion. Raises the DiscussionWasStarted event.
      */
-    public static function start(?string $title, User $user, self $model = null): self
+    public static function start(?string $title, User $user, ?self $model = null): self
```

Comment:
Optional arguments should always be listed after required ones. I know this is going to be non-backward compatible though. Not sure if we want to take care of that now, but it should be documented at least.

But considering this PR is going to be break every extension that introduce their own extenders, we might as well tackle this in one swoop too?

Reply:
@luceos this is not a breaking change, model is the only optional argument and it is provided last, a nullable argument is not an optional argument 👍🏼

Reply:
> @luceos this is not a breaking change, model is the only optional argument and it is provided last, a nullable argument is not an optional argument 👍🏼

I missed that, good call.